packages/textInput/components/TextInput.tsx

```diff
@@ -177,6 +192,29 @@ export class TextInput<
       />
     );
   }
+
+  protected getTooltipContent(): React.ReactNode {
+    if (!this.props.tooltipContent) {
+      return null;
+    }
+
+    return (
+      <span className={cx(margin("left", "xs"))}>
+        <Tooltip
+          trigger={
+            <Icon
+              color={greyLightDarken2}
+              shape={SystemIcons.CircleQuestion}
+              size={iconSizeXs}
+            />
+          }
+          id="tooltipTextinput"
```

Comment:
We'll get duplicate IDs if we just use a string for this. We could use the `id` prop to make the tooltip IDs unique